sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/DefaultServiceSpec.java

```diff
@@ -84,6 +84,10 @@ public DefaultServiceSpec(
         this.role = role;
         this.principal = principal;
         this.user = getUser(user, pods);
+        this.goalState = goalState == null ? GoalState.RUNNING : goalState;
+        if (goalState == GoalState.FINISHED) {
+            throw new IllegalArgumentException("Service goal state is deprecated FINISHED. Did you mean FINISH?");
```

Comment:
Doesn't a goal state of `ONCE` really make more sense at the service level when it's representing a job?

Reply:
Not sure, I just went by the comments in `GoalState`:

```
    /**
     * Running and then exiting successfully. Should not be restarted upon a successful exit.
     */
    FINISH,

    /**
     * Running only once over the lifetime of a service. Should not be run again for the duration of this install.
     */
    ONCE
```